pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -212,6 +216,63 @@ protected void doShutdown() {
     if (_leaseExtender != null) {
       _leaseExtender.shutDown();
     }
+    // Make sure we do metric cleanup when we shutdown the table.
+    if (_maximumPinotConsumptionDelayTracker != null) {
+      _maximumPinotConsumptionDelayTracker.shutdown();
+    }
+    // Now that segments can't report metric destroy metric for this table
+    _serverMetrics.removeTableGauge(_tableNameWithType, ServerGauge.MAX_PINOT_CONSUMPTION_DELAY_MS);
+  }
+
+  /*
+   * Pass-through method to handle CONSUMING->DROPPED transition.
+   *
+   * @param pid Partition id that we must stop tracking on this server.
+   */
+  public void stopTrackingPartitionDelay(long pid) {
+    if (_maximumPinotConsumptionDelayTracker != null) {
+      _maximumPinotConsumptionDelayTracker.stopTrackingPinotConsumptionDelayForPartition(pid);
+    }
+  }
+
+  /*
+   * Pass-through method to handle CONSUMING->ONLINE transition.
```

Comment:
suggest remove this comment line (and other similar lines)

Reply:
done